### PR TITLE
Meerkatplus

### PR DESCRIPTION
**Pull request by mattieudv**
Added features required for meerkat+ usage, and backwards compatible with meerkat. Done tests on laptop with an old h5 file, and also some lab tests.

### CODE REVIEW

File: scripts/pythonclient.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from websocket import create_connection
 # ws = create_connection("ws://10.98.2.128:31460/ws")
+ws = create_connection("ws://10.98.2.128:31020/ws")
```
Comment (ludwigschwardt) [inline scripts/pythonclient.py:3]: Is this a fixed port number?

File: scripts/time_plot.py [hunk 2]
```diff
@@ -435,7 +435,7 @@ def RingBufferProcess(multicast_group, spead_port, spead_interface, memusage, ma
                         if (len(signal)<len(ts)):
                             signal=np.r_[signal,np.tile(np.nan,len(ts)-len(signal))]
                         ydata.append(signal)#should check that correct corresponding values are returned
-                        legend.append(datasd.cpref.id_to_real_str(id=product,short=True).replace('m00','').replace('m0','').replace('m','').replace('ant','').replace(' * ',''))
+                        legend.append(datasd.cpref.id_to_real_str(id=product,short=True).replace('s000','s').replace('s00','s').replace('s0','s').replace('m00','').replace('m0','').replace('m','').replace('ant','').replace(' * ',''))
```
Comment (ludwigschwardt) [inline scripts/time_plot.py:438]: A regex would be nicer :-)

File: scripts/time_plot.py [hunk 3]
```diff
@@ -2181,13 +2181,29 @@ def handle_websock_event(handlerkey,*args):
...
     sreg=re.compile('[h|v|H|V|x|y]').split(signalstr)
     if (len(sreg)!=3 or len(sreg[2])!=0):
-        return ();
-    if ((not sreg[0].isdigit()) or (not sreg[1].isdigit())):
-        return (sreg[0]+signalstr[len(sreg[0])],sreg[1]+signalstr[len(sreg[0])+1+len(sreg[1])])
-    return (ANTNAMEPREFIX%(int(sreg[0]))+signalstr[len(sreg[0])].lower(),ANTNAMEPREFIX%(int(sreg[1]))+signalstr[len(sreg[0])+1+len(sreg[1])].lower())
+        return ()
+    if sreg[0].isdigit():
+        rv0='m%03d'%int(sreg[0])
+    elif sreg[0][0].lower()=='m' and sreg[0][1:].isdigit():
```
Comment (ludwigschwardt) [inline scripts/time_plot.py:2196]: More things that beg for regular expressions...

File: scripts/time_plot.py [hunk 4]
```diff
@@ -1493,7 +1494,7 @@ def handle_websock_event(handlerkey,*args):
                     else:
                         send_websock_cmd('logconsole("Invalid reference antenna specified, using default instead",true,true,true)',handlerkey)
                         refantnumber=antnumbers[0]
-                send_websock_cmd('logconsole("Building waterfall table for: '+','.join(['m%03d'%antnum for antnum in antnumbers])+'",true,false,true)',handlerkey)
+                send_websock_cmd('logconsole("Building waterfall table for: '+','.join(np.r_[['m%03d'%antnum for antnum in antnumbers],skaants])+'",true,false,true)',handlerkey)
```
Comment (ludwigschwardt) [inline scripts/time_plot.py:1497]: If `skaants` is a list (I think it is), it is cleaner to just do:
```python
','.join(['m%03d'%antnum for antnum in antnumbers] + skaants)
```

**Review by ludwigschwardt — APPROVED** [review]
Some minor suggestions. You could do at least the list addition and leave the regexes for another time :-)

File: scripts/time_plot.py (hunk 3 above)
Comment (mattieudv) [reply scripts/time_plot.py:2196]: Will skip regex for now. I think other instances where I did use regex is a bit less readable, although I admit this seems a bit clumsy.

**Pull request merged**